validation/helpers.py

```diff
@@ -175,27 +180,65 @@ def get_translations_from_itwewina(word):
         return "Error getting request"
 
 
+def get_translations(results, from_lemma=False):
+    matches = []
+
+    for i in results["results"]:
+        translations, analysis = extract_translations(i["lemma_wordform"])
+        if from_lemma:
+            translations = "(Lemma translation) " + translations
+        if {"translations": translations, "analysis": analysis} not in matches:
+            matches.append({"translations": translations, "analysis": analysis})
+
+    return matches
+
+
+def extract_translations(entry):
+    translations = []
+
+    if type(entry["definitions"]) == list and len(entry["definitions"]) > 0:
+        translations = [str(j["text"]) for j in entry["definitions"]]
+
+    translations = "; ".join(translations)
+    analysis = entry["analysis"]
+
+    return translations, analysis
+
+
+def get_lemma(word):
```

Comment:
This is very similar to @eddieantonio’s https://gist.github.com/eddieantonio/8f3a678e69f0595dfd1a1e42239e68e0#file-crk_analyze-py-L79-L93 ; if you can wait a bit, we can make a library version to call, though that might not happen until tomorrow

Reply:
I can wait for the updated library. There's still quite a bit of refactoring in this branch that will need to be reviewed, but we might as one do one merge with the proper calls to `hfst-optimized-lookup`

Reply:
@nienna73 Upgrade to `hfst-optimized-lookup==0.0.10` and then you can do something like this!

```suggestion

fst = TransducerFile("crk-strict-analyzer-for-dictionary.hfstol")

def get_lemma(word):
    try:
        analysis, = fst.lookup_lemma_with_affixes(word)
    except ValueError as err:
        raise ValueError(f"more than one analysis for {word}") from err
    return analysis.lemma
```